core/src/test/java/org/sonarsource/sonarlint/core/container/connected/update/ModuleHierarchyDownloaderTest.java

```diff
@@ -69,9 +72,59 @@ public void test() {
   }
 
   @Test
+  public void testNoPagination() throws IOException {
+    TreeWsResponse.Builder responseBuilder = TreeWsResponse.newBuilder()
+      .setPaging(Paging.newBuilder().setTotal(500));
+    for (int i = 0; i < 500; i++) {
```

Comment:
Much less than 500 iterations could equally prove a point (in `testPagination` too, further below)